src/middleware/jira-symmetric-jwt-middleware.ts

```diff
@@ -5,6 +5,31 @@ import { getJWTRequest, TokenType, validateQsh } from "~/src/jira/util/jwt";
 import { Installation } from "~/src/models/installation";
 import { moduleUrls } from "~/src/routes/jira/atlassian-connect/jira-atlassian-connect-get";
 import { matchRouteWithPattern } from "~/src/util/match-route-with-pattern";
+import { getJiraClient } from "~/src/jira/client/jira-client";
+
+export const setJiraAdminPrivileges = async (req: Request, claims, jiraHost, installationId) => {
+	// We only need to add this to the session if it doesn't exist
+	if (req.session.isJiraAdmin !== undefined) {
+		return;
+	}
+	try {
+		const userAccountId = claims.sub;
+		const jiraClient = await getJiraClient(
+			jiraHost,
+			installationId,
+			undefined
+		);
+
+		// Make jira call to permissions with userAccountId.
+		const permissions = await jiraClient.permissions.checkAdmin(userAccountId);
+		const hasAdminPermissions = permissions.data.globalPermissions.includes("ADMINISTER");
+
+		req.session.isJiraAdmin = hasAdminPermissions;
```

Comment:
How do we refresh this var? Say 1) user figure our they can't do something in our app, 2) they are added as admin 3) they come back still can't do thing since the session contains `isAdmin: fasle`

Reply:
ha, yeah I considered this one, then kinda thought it would be ok, they need to start a new session so a fresh browser would work i think, i dunno about a tab? but it is a pretty esoteric way to get it working.

I might be able to set it against the jira login token so they would still need to log out and log in but I think thats pretty standard for perm changes anyway

Reply:
So.. its bit of a nuisance for this is refresh, I dont want to check everytime as its a bit expensive time wise.

What we can do though, we are going to present failed/rejected users with some sort of messaging. we can tell them there to start it in a fresh login/browser??

Reply:
Hmm, actually, I think we might be able to even avoid putting that `isJiraAdmin` into session.
Can you check in our normal workflow within the app (not the create branch), the jwt token will it contains user id? 
If no, then there's no need to put it in session. Coz in create branch we don't check the admin permission. So not much place we really need to check admin permission because this code return earlier if userAccountId is empty.

Reply:
Create branch has user id on it, just double checked.